src/index.ts

```diff
@@ -37,12 +37,12 @@ export const options: Record<
         default: [{ value: ['typescript', 'jsx'] }],
         description: 'Provide a list of plugins for special syntax',
     },
-    importOrderCombineTypeAndValueImports: {
-        type: 'boolean',
+    importOrderTypeScriptVersion: {
+        type: 'string',
         category: 'Global',
-        default: false,
+        default: '1.0.0',
```

Comment:
*Should* we default to 1.0.0?

This is a breaking change, so we could default to 5.0.0 instead

Reply:
https://github.com/IanVS/prettier-plugin-sort-imports/issues/22#issuecomment-1522049180

We could alternately not default unless we can't detect :P

Reply:
I responded.  I think that there are risks we'll detect the wrong version, but I think I'm okay with that as long as we keep this option as an override / escape hatch in case the auto-detection is incorrect.  

What do you think about running with just the explicit version for right now, and adding in auto-detection in another PR?

Reply:
No objections.

I just wanted to advocate for the thing that helps people that are starting new projects! That's likely the first spike of interest in a tool like this, followed by new people joining mature (or stagnant) complex codebases.

I think that if people that are stuck on old versions, that's not a surprise to them, and they will notice the "1 other option" that tells them to pass their TypeScript version if they have a complex setup!

I also think it's unlikely that typescript will drastically change import syntax in the future -- it has to stay in sync with the wider JS Community there.

Anyhow, feel free to proceed without auto-detection now. Technically that can always be added in a future release. Technically a breaking behavior change, but… I wouldn't follow SemVer when changing that -- just my 2 cents.